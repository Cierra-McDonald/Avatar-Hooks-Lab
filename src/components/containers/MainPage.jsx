import React, {useState, useEffect } from 'react'
import { getAvatars } from '../services/AvatarApi'
import CharacterList  from './CharacterList'

const MainPage = () => {
    const [avatars, setAvatars] = useState([]);
    const [loading, setLoading] = useState(true);

    useEffect(() => {
        getAvatars()
            .then(characters => {
                setAvatars(characters)
            })
            .finally(() => setLoading(false));

    }, [])

    return(
        <div>
            <CharacterList 
                avatars={avatars}
                loading={loading}
            />
        </div>
    )
    
}

export default MainPage;
