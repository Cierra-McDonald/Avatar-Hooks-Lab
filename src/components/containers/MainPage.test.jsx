import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter} from 'react-router'
import  MainPage  from './MainPage'

describe('Testing MainPage Container', () => { 
    // beforeAll(() => server.listen());
    // afterAll(() => server.close());

    it('should render a list of avatar characters and redired to a detail page', async () => { 
        render(<MemoryRouter> <MainPage/> </MemoryRouter>);

        screen.getByText('Loading...')

        const ul = await screen.findByRole('list', {name: 'avatars'})
        expect(ul).toMatchSnapshot();
    });
});