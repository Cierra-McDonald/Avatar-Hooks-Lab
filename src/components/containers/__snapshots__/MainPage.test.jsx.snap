// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MainPage Container should render a list of avatar characters and redired to a detail page 1`] = `
<ul
  aria-label="avatars"
  class="ul"
>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Amon
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab68d0"
      >
        <img
          alt="Amon"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/0/00/Amon_and_his_Equalists.png/revision/latest?cb=20160219113422"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
         Equalists (formerly) Northern Water Tribe (formerly)
      </p>
      <p>
         Allies: 
        Tarrlok
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Arnook
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab68d6"
      >
        <img
          alt="Arnook"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/0/0c/Chief.png/revision/latest?cb=20140122221730"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
        Northern Water Tribe
      </p>
      <p>
         Allies: 
        Aang
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Bato
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab68e7"
      >
        <img
          alt="Bato"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/e/e3/Bato_smiling.png/revision/latest?cb=20130626122901"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
        Southern Water Tribe
      </p>
      <p>
         Allies: 
        Aang
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Desna and Eska
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab6912"
      >
        <img
          alt="Desna and Eska"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/1/1b/Chiefs_Desna_and_Eska.png/revision/latest?cb=20140730174236"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
        Northern Water Tribe
      </p>
      <p>
         Allies: 
        Northern Water Tribe
        
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Gilak
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab6941"
      >
        <img
          alt="Gilak"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/c/cd/Gilak.png/revision/latest?cb=20170524065046"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
        Southern Water Tribe
      </p>
      <p>
         Allies: 
        Hakoda
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Hahn
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab6953"
      >
        <img
          alt="Hahn"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/4/46/Hahn.png/revision/latest?cb=20140122221732"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
        Northern Water Tribe
      </p>
      <p>
         Allies: 
        Northern Water Tribe
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Hakoda
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab6954"
      >
        <img
          alt="Hakoda"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/4/4e/Hakoda_smiling.png/revision/latest?cb=20140510202641"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
        Southern Water Tribe
      </p>
      <p>
         Allies: 
        Kya
        
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Hama
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab6956"
      >
        <img
          alt="Hama"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/4/4d/Sinister_Hama.png/revision/latest?cb=20111215082556"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
        Southern Water Tribe
      </p>
      <p>
         Allies: 
        Southern Water Tribe
        
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Hiryu
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab6966"
      >
        <img
          alt="Hiryu"
          class="mainPic"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
        Northern Water Tribe
      </p>
      <p>
         Allies: 
        Team Avatar
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Hotah
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab696a"
      >
        <img
          alt="Hotah"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/4/4c/Judge_Hotah.png/revision/latest?cb=20131008135523"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
        Northern Water Tribe
      </p>
      <p>
         Allies: 
        Unalaq
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Iknik Blackstone Varrick
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab6974"
      >
        <img
          alt="Iknik Blackstone Varrick"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/4/48/Varrick_looking_smug.png/revision/latest?cb=20131116223858"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
         Kuvira's army (deserted) Metal Clan (formerly) Southern Water Tribe rebels (formerly) Team Avatar Varrick Global Industries
      </p>
      <p>
         Allies: 
        Zhu Li
        
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Kanna
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab6990"
      >
        <img
          alt="Kanna"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/4/47/Kanna.png/revision/latest?cb=20130822224809"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
         Northern Water Tribe (formerly) Southern Water Tribe
      </p>
      <p>
         Allies: 
        Southern Water Tribe
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Katara
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab6992"
      >
        <img
          alt="Katara"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/7/7a/Katara_smiles_at_coronation.png/revision/latest?cb=20150104171449"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
         Team Avatar Water Tribe
      </p>
      <p>
         Allies: 
        Southern Water Tribe
        Aang
        
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Korra
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab69a1"
      >
        <img
          alt="Korra"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/c/ca/Korra.png/revision/latest?cb=20150406235047"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
         Fire Ferrets (formerly) Southern Water Tribe Tarrlok's task force (formerly) Team Avatar
      </p>
      <p>
         Allies: 
        Southern Water Tribe
        
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Kya
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab69a9"
      >
        <img
          alt="Kya"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/7/77/Kya_%28waterbender%29.png/revision/latest?cb=20151102093816"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
        Southern Water Tribe
      </p>
      <p>
         Allies: 
        Southern Water Tribe
        
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Kya (nonbender)
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab69aa"
      >
        <img
          alt="Kya (nonbender)"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/b/b5/Kya_%28nonbender%29.png/revision/latest?cb=20170523213135"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
        Southern Water Tribe
      </p>
      <p>
         Allies: 
        Southern Water Tribe
        
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Kya (pilot)
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab69ab"
      >
        <img
          alt="Kya (pilot)"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/d/de/Kya_pilot.png/revision/latest?cb=20130126192101"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
         Team Avatar Water Tribes
      </p>
      <p>
         Allies: 
        Aang
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        La
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab69ad"
      >
        <img
          alt="La"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/3/32/La.png/revision/latest?cb=20140124171520"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
         Spirit World Water Tribe
      </p>
      <p>
         Allies: 
        Tui
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Malina
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab69ca"
      >
        <img
          alt="Malina"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/8/84/Malina.png/revision/latest?cb=20161026183603"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
        Southern Water Tribe
      </p>
      <p>
         Allies: 
        Hakoda
        
      </p>
    </div>
  </li>
  <li
    class="list"
  >
    <div
      class="mainCards"
    >
      <p>
        Character: 
        Maliq
      </p>
      <a
        activeclassname="selected"
        exact="true"
        href="/details/5cf5679a915ecad153ab69cb"
      >
        <img
          alt="Maliq"
          class="mainPic"
          src="https://vignette.wikia.nocookie.net/avatar/images/4/42/Maliq.png/revision/latest?cb=20161026183604"
          style="width: 250px; border-radius: 5em;"
        />
      </a>
      <p>
         Affiliation: 
         Ba Sing Se University[1] (formerly) Northern Water Tribe
      </p>
      <p>
         Allies: 
        Malina
      </p>
    </div>
  </li>
</ul>
`;
